warehouse/upload.go

```diff
@@ -912,10 +911,14 @@ func (job *UploadJobT) getTotalCount(tName string) (int64, error) {
 	expBackoff.RandomizationFactor = 0
 	expBackoff.Reset()
 	backoffWithMaxRetry := backoff.WithMaxRetries(expBackoff, 5)
-	err := backoff.RetryNotify(operation, backoffWithMaxRetry, func(err error, t time.Duration) {
-		pkgLogger.Errorf(`Error getting total count in table:%s error: %v`, tName, err)
+	retryError := backoff.RetryNotify(operation, backoffWithMaxRetry, func(err error, t time.Duration) {
+		err = fmt.Errorf(`Error getting total count in table:%s error: %v`, tName, err)
+		pkgLogger.Error(err)
 	})
-	return total, err
+	if retryError != nil {
```

Comment:
Directly return retryError here instead of having this check?

Reply:
There are 2 reasons why we can't do this:
1. We have already declared (total, err) at the function declaration.
2. We can't return `retryError` because it is the error for the backoff.RetryNotify function.

Reply:
Arent we returning the same retryError after assigning to err?